# IntegrativeNetworkMap
We generated a maize network map that integrates genomic interactions, transcriptomic co-expression networks, translatomic co-translation networks and protein-protein interactions across maize development. We deposited all the raw data in the NCBI (https://www.ncbi.nlm.nih.gov/geo/) under GEO accession number GSE199932, which will be released after the manuscript is accepted.  
##### All analysis was divided into 6 sections:  
##### Section0: Identification and quantification of mRNAs, lncRNAs, circRNAs and small RNAs; calculation of protein-coding RNA expression by Ribo-seq; and acquisition of protein-protein interactions.
##### Section1: Construction of six different type networks (Slim co-expression network, Co-translation network, Co-expression network with ncRNA, Interactome, Slim-IntegrativeOmics, IntegrativeOmics with ncRNAs) and statistics of network properties.
##### Section2: Clustering and assessing functional divergence of duplicated genes from the network perspective.
##### Section3: Asymmetric regulatory divergence between the two maize subgenomes.
##### Section4: Prediction gene function and reconstruction regulating networks underlying complex traits.
##### Section5: The integrative network map boosts prediction of FT-genes by machine learning.
